src/lib/bundle-serve.js

```diff
@@ -22,39 +23,62 @@ module.exports = async (bundler, options, log = () => { }, _actionConfig) => {
   process.env.__OW_API_KEY = process.env.AIO_runtime_auth
   process.env.__OW_NAMESPACE = process.env.AIO_runtime_namespace
 
-  // options.dist ??
   const cert = fs.readFileSync(options.https.cert, 'utf-8')
   const key = fs.readFileSync(options.https.key, 'utf-8')
   const serverOptions = {
     key,
     cert
   }
+  let subscription
 
   try {
-    let { bundleGraph, buildTime } = await bundler.run()
-    let bundles = bundleGraph.getBundles()
+    // run it once
+    const { bundleGraph, buildTime } = await bundler.run()
+    const bundles = bundleGraph.getBundles()
     console.log(`✨ Built ${bundles.length} bundles in ${buildTime}ms!`)
+
+    subscription = await bundler.watch((err, event) => {
+      if (err) {
+        // fatal error
+        throw err
+      }
+
+      aioLogger.info(`${event.changedAssets.size} static asset(s) changed`)
+      const limit = options.verbose ? Infinity : 5
+      if (event.changedAssets.size <= limit) {
+        event.changedAssets.forEach((value, key, map) => {
+          aioLogger.info('\t-->', value)
+        })
+      }
+      if (event.type === 'buildSuccess') {
+        const bundles = event.bundleGraph.getBundles()
+        aioLogger.info(`✨ Built ${bundles.length} bundles in ${event.buildTime}ms!`)
+      } else if (event.type === 'buildFailure') {
+        aioLogger.error(event.diagnostics)
+      }
+    })
   } catch (err) {
-    console.log(err.diagnostics)
+    aioLogger.error(err.diagnostics)
   }
 
   const app = express()
   app.use(express.json())
-  app.use(express.static('dist/dx-excshell-1/web-dev'))
+  app.use(express.static(options.dist))
   // DONE: serveAction needs to clear cache for each request, so we get live changes
   app.all('/api/v1/web/*', serveAction)
 
-  const port = options.serveOptions.port || Number(process.env.PORT || 9000)
+  const port = options.port || Number(process.env.PORT || 9000)
   const server = https.createServer(serverOptions, app)
   server.listen(port, () => {
-    console.log('server running on port : ' + port)
+    aioLogger.info('server running on port : ' + port)
   })
-  const url = `${options.serveOptions.https ? 'https:' : 'http:'}//localhost:${port}`
+  const url = `${options.https ? 'https:' : 'http:'}//localhost:${port}`
 
   const serverCleanup = async () => {
-    console.debug('shutting down server ...')
+    aioLogger.info('shutting down server ...')
     await app.close()
     await server.close()
+    subscription ?? await subscription.unsubscribe()
```

Comment:
not sure what I was thinking - this is wrong

Reply:
fixed in https://github.com/adobe/aio-cli-plugin-app-dev/pull/14/commits/960358df0b259c5f796e6e980d91374e58b20cea